crates/rattler_conda_types/src/match_spec/mod.rs

```diff
@@ -186,12 +206,33 @@ impl MatchSpec {
             }
         }
 
+        if let Some(md5_spec) = self.md5.as_ref() {
+            if !record
+                .md5
+                .as_ref()
+                .map_or(false, |md5_record| md5_spec == md5_record)
```

Comment:
You can use something along the lines of Some(md5_spec) == record.md5.as_ref() to simplify the code a bit.